server/game/GameActions/DelayedEffectAction.ts

```diff
@@ -21,9 +22,18 @@ export class DelayedEffectAction extends CardGameAction {
         super(properties);
     }
 
-    canAffect(card: BaseCard, context: AbilityContext): boolean {
-        if(card.location !== Locations.PlayArea) {
-            return false;
+    canAffect(card: BaseCard, context: AbilityContext, additionalProperties = {}): boolean {
+        let properties = this.getProperties(context, additionalProperties) as DelayedEffectActionProperties;
+        if(properties.location) {
+            let inLocation = Array.isArray(properties.location) ? properties.location.includes(card.location) : properties.location === card.location;
+            if(!inLocation) {
+                return false;
+            }
+        }
+        else {
+            if(card.location !== Locations.PlayArea) {
+                return false;
+            }
```

Comment:
There's a number of things about the way you've done this that I don't really like (but it could just be personal preference):
 - For what you're trying to do, I'd rather see you using `defaultProperties` than hard coding default behaviour like this
 - Your location property will get passed to the DelayedEffect too - I'm not sure that an Array is the right thing to pass here, maybe it should be the current location of the card?